core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataCutter.scala

```diff
@@ -123,14 +126,41 @@ class DataCutter(uid: String = UID[DataCutter]) extends Splitter(uid = uid) with
       .filter(r => labelSet.contains(r.getDouble(labelColIdx)))
       .withColumn(labelColName, data(labelColName).as(labelColName, metadataNA.toMetadata))
 
+    // calculate the down sample fraction
+    val dataSetSize = data.count()
+    val sampleF = getMaxTrainingSample / dataSetSize.toDouble
+    val downSampleFraction = math.min(sampleF, SplitterParamsDefault.DownSampleFractionDefault)
+    setDownSampleFraction(downSampleFraction)
+
     summary = Option(DataCutterSummary(
+      preSplitterDataCount = dataSetSize,
+      downSamplingFraction = getDownSampleFraction,
       labelsKept = getLabelsToKeep,
       labelsDropped = getLabelsToDrop,
       labelsDroppedTotal = getLabelsDroppedTotal
     ))
     PrevalidationVal(summary, Option(dataPrep))
```

Comment:
I think it makes more sense to do the downsampling in the pre-validationPrepare than in the validation prepare - the difference being that the validation prepare is called within the CV folds. So since you upsample for binary it needs to be here to prevent label leakage but since there is only downsampling here it can go earlier

Reply:
The 2 main reasons behind doing it this way:
1. Keeping it consistent with dataSplitter/dataBalancer
2. Easier to implement stratified sampling (or other data balancing techniques) in the future which would upsample the minority classes